src/ScriptEngine/ScriptEngine.cs

```diff
@@ -114,60 +121,88 @@ private void LoadDLL(string path, GameObject obj)
             if (!QuietMode.Value)
                 Logger.Log(LogLevel.Info, $"Loading plugins from {path}");
 
-            using (var dll = AssemblyDefinition.ReadAssembly(path, new ReaderParameters { AssemblyResolver = defaultResolver }))
+            using (var dll = AssemblyDefinition.ReadAssembly(path, new ReaderParameters {
+                AssemblyResolver = defaultResolver,
+                ReadSymbols = true
+            }))
             {
                 dll.Name.Name = $"{dll.Name.Name}-{DateTime.Now.Ticks}";
+                Assembly ass;
 
-                using (var ms = new MemoryStream())
+                if (DumpAssemblies.Value)
                 {
-                    dll.Write(ms);
-                    var ass = Assembly.Load(ms.ToArray());
+                    // Dump assembly & load it from disk
+                    if (!Directory.Exists(DumpedAssembliesPath))
+                        Directory.CreateDirectory(DumpedAssembliesPath);
+   
+                    string assemblyDumpPath = Path.Combine(DumpedAssembliesPath, dll.Name.Name + Path.GetExtension(dll.MainModule.Name));
```

Comment:
Why the Path.GetExtension instead of just adding .dll?

Reply:
I copied that from the Preloader. I assume it might be relevant on non-Windows OSes